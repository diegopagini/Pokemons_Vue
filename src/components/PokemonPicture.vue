<!-- @format -->

<template>
	<div class="pokemon-container">
		<div class="pokemon-wrapper">
			<img
				alt="pokemon"
				class="hidden-pokemon"
				:src="imgSrc"
			/>

			<img
				v-if="showPokemon"
				alt="pokemon"
				class="fade-in"
				:src="imgSrc"
			/>
		</div>
	</div>
</template>

<script>
	export default {
		props: {
			pokemonId: {
				type: Number,
				required: true,
			},
			showPokemon: {
				type: Boolean,
				required: true,
				default: false,
			},
		},
		computed: {
			imgSrc() {
				return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`;
			},
		},
	};
</script>

<style scoped>
	.pokemon-container {
		height: 200px;
	}

	.pokemon-wrapper {
		display: flex;
		justify-content: center;
		position: relative;
		width: 100%;
	}

	img {
		-moz-user-select: none;
		-ms-user-select: none;
		-webkit-user-drag: none;
		-webkit-user-select: none;
		height: 200px;
		position: absolute;
		user-select: none;
	}
	.hidden-pokemon {
		filter: brightness(0);
	}
</style>
