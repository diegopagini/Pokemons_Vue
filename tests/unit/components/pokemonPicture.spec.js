/** @format */
import PokemonPicture from '@/components/PokemonPicture';
import { shallowMount } from '@vue/test-utils';

describe('PokemonPicture component', () => {
	test('debe de hacer match con el snapshot', () => {
		const wrapper = shallowMount(PokemonPicture, {
			props: {
				pokemonId: 1,
				showPokemon: false,
			},
		});
		expect(wrapper.html()).toMatchSnapshot();
	});

	test('debe de mostrar la imagen oculta y el pokemon 100', () => {
		const wrapper = shallowMount(PokemonPicture, {
			props: {
				pokemonId: 100,
				showPokemon: false,
			},
		});
		const { pokemonId } = wrapper.props();
		expect(pokemonId).toEqual(100);
		const [img1, img2] = wrapper.findAll('img');
		expect(img1.exists()).toBeTruthy();
		expect(img2).toBeUndefined();
		expect(img1.classes('hidden-pokemon')).toBeTruthy();
		expect(img1.attributes('src')).toContain('100');
	});

	test('debe de mostrar el pokemon si el showPokemon:true', () => {
		const wrapper = shallowMount(PokemonPicture, {
			props: {
				pokemonId: 100,
				showPokemon: true,
			},
		});
		const img = wrapper.find('img');
		expect(img.exists()).toBeTruthy();
		expect(img.classes('hidden-pokemon')).toBe(true);
	});
});
