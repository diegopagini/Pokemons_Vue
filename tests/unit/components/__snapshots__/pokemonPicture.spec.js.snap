// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture component debe de hacer match con el snapshot 1`] = `
<div class="pokemon-container">
  <div class="pokemon-wrapper"><img alt="pokemon" class="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg">
    <!--v-if-->
  </div>
</div>
`;
